GREEDY/DataManagers/ImageFormating.cs

```diff
@@ -1,16 +1,154 @@
-﻿using System.Drawing;
+﻿using System;
+using System.Collections.Generic;
+using System.Drawing;
+using AForge;
+using AForge.Imaging.Filters;
 using Emgu.CV;
+using Emgu.CV.CvEnum;
 using Emgu.CV.Structure;
+using Emgu.CV.Util;
+using Point = System.Drawing.Point;
 
-namespace GREEDY.Services
+namespace GREEDY.DataManagers
 {
     public class ImageFormating : IImageFormating
     {
-        public Bitmap Blur(Bitmap bitmap, int width, int height)
+        private Bitmap BradleyLocalThreshold(Bitmap bitmap)
         {
             Image<Gray, byte> inputImage = new Image<Gray, byte>(bitmap);
-            Image<Gray, float> blurredImage = Image<Gray, float>.FromIplImagePtr(inputImage.Ptr).SmoothBlur(width, height);
-            return blurredImage.Bitmap;
+            BradleyLocalThresholding bradleyLocalThresholding = new BradleyLocalThresholding();
+            Bitmap filtered = new Bitmap(bradleyLocalThresholding.Apply(inputImage.Bitmap));
+            return filtered;
+        }
+
+        private Bitmap Crop(Bitmap bitmap, double[] cornerPoints)
+        {
+            // define quadrilateral's corners
+            List<IntPoint> corners = new List<IntPoint>
+            {
+                new IntPoint(Convert.ToInt32(cornerPoints[0]), Convert.ToInt32(cornerPoints[1])),
+                new IntPoint(Convert.ToInt32(cornerPoints[2]), Convert.ToInt32(cornerPoints[3])),
+                new IntPoint(Convert.ToInt32(cornerPoints[4]), Convert.ToInt32(cornerPoints[5])),
+                new IntPoint(Convert.ToInt32(cornerPoints[6]), Convert.ToInt32(cornerPoints[7]))
+            };
+            QuadrilateralTransformation filter = new QuadrilateralTransformation(corners);
+            Bitmap newImage = filter.Apply(bitmap);
+            return newImage;
+        }
+
+        //returns the bigest found rectangle
+        private Bitmap FilterCropedImages(List<Bitmap> list)
+        {
+            Bitmap correctImage = null;
+            foreach (Bitmap img in list)
+            {
+                if (correctImage == null)
+                    correctImage = img;
+                else if ((img.Height * img.Width) > (correctImage.Height * correctImage.Width))
+                    correctImage = img;
+            }
```

Comment:
I believe this method can be rewritten with linq using the MoreLinq library. Then this whole method would be one line somthing like: return list.MaxBy(x => x.Height * x.Width); It would be more readable and more effective aswell.